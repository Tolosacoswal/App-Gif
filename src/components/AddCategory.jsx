import { useState } from "react"
import { propTypes } from "prop-types";

export const AddCategory=({onNewCategory})=>{
     const [inputValue, setInputValue] = useState(``);
    const onInpuntChange = ({target})=> {
    setInputValue (target.value);
    }
    const onSubmit = (event)=> {
        
            event.preventDefault();
           if (inputValue.trim().length <=1)return;
            // setCategories (categories=>[inputValue,...categories]);
           setInputValue(""); 
           onNewCategory( inputValue.trim());
            
        }
    return (
        <form onSubmit={onSubmit} aria-label="form" >
            <input
        type="text"
        placeholder="Buscar GIF"
        value= {inputValue}
        onChange={ onInpuntChange }
        />
        </form>
        
    )
    AddCategory.propType ={
        onNewCategory: propTypes.func.isRequired,
        }
    }
