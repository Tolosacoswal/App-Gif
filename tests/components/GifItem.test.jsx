import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('pruebas en <GifItem />', () => { 
    const title = `luffy`;
    const url ="https://one-piece.com/luffy.jpg"
test('debe hacer match con el snapshot', () => {

const {container}=render(<GifItem title={title} url={url}/>);
expect (container).toMatchSnapshot();
})
 })