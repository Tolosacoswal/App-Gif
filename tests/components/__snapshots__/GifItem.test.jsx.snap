// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="luffy"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
      luffy
    </p>
  </div>
</div>
`;
